<launch>
    <arg name="config_path" default = "$(find ncrl_lio)/config/run.yaml" />

    <arg name="rviz" default="true"/>
    <group if="$(arg rviz)">
        <param name="use_sim_time" value="true"/>
        <node pkg="rviz" type="rviz" name="rviz" args="-d $(find ncrl_lio)/rviz_cfg/run.rviz"/>
    </group>

    <!-- node 1 & 2 feature extraction and frame to frame odometry estimate -->
    <arg name="aloam_modify" default="true"/>
    <group if="$(arg aloam_modify)">
        <param name="minimum_range" type="double" value="0.5"/>
        <param name="scan_line" type="int" value="16" />
        <param name="mapping_skip_frame" type="int" value="1"/>
<!--        <remap from="/velodyne_points" to="/points_raw"/>-->
        <node pkg="ncrl_lio" type="scanRegistration" name="scanRegistration" launch-prefix="taskset -c 3"/>
        <node pkg="ncrl_lio" type="laserOdometry" name="laserOdometry" launch-prefix="taskset -c 2"/>
    </group>

    <!-- node 3 : estimate state and perform tightly coupled -->
    <group if="true">
       <param name="config_file" type="string" value="$(arg config_path)" />
        <node pkg="ncrl_lio" type="ncrl_lio_estimator_node_fusion" name="ncrl_lio_estimator_node_fusion" output="screen" launch-prefix="taskset -c 1"/>
    </group>

</launch>
